sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -300,42 +300,35 @@ def __init__(self, data_stub):
 class GrpcServerDataChannel(
     beam_fn_api_pb2_grpc.BeamFnDataServicer, _GrpcDataChannel):
   """A DataChannel wrapping the server side of a BeamFnData connection."""
-
-  def Data(self, elements_iterator, context):
-    self._start_reader(elements_iterator)
-    for elements in self._write_outputs():
-      yield elements
-
-
-class GrpcServerMultiPlexDataChannel(GrpcServerDataChannel):
-  """A DataChannel wrapping the server side of a BeamFnData connection,
-  implementing data multiplexing."""
-
-  def __init__(self, task_worker_mapping):
+  def __init__(self):
     super(GrpcServerDataChannel, self).__init__()
-    self._to_send_per_worker = collections.defaultdict(queue.Queue)
-    self._task_worker_mapping = task_worker_mapping
+    self._conn_handler = ConnectionHandler()
 
   def close(self):
-    for q in self._to_send_per_worker.values():
-      q.put(self._WRITES_FINISHED)
+    for conn_handler in self._conn_handler.get_conn_handlers().values():
+      conn_handler.add(self._WRITES_FINISHED)
     self._closed = True
+    self._conn_handler.close()
 
   def output_stream(self, instruction_id, transform_id):
     def add_to_send_queue(data):
       if data:
-        worker_id = self._task_worker_mapping[instruction_id]
-        self._to_send_per_worker[worker_id].put(
+        worker_id = 'worker_%s' % (int(instruction_id.split('_')[1]) %
```

Comment:
It seems brittle to depend on the way workers are named here. Instead, split things up as with the Control connection, putting the bulk of the logic (including methods like `output_stream` and `_write_outputs`) onto the `ChannelConnection` connection object that already only holds a connection to the right worker.